pkg/lib/aws/s3.go

```diff
@@ -429,6 +430,39 @@ func (c *Client) DeletePrefix(bucket string, prefix string, continueIfFailure bo
 	return nil
 }
 
+func (c *Client) CreateBucket(bucket string) error {
+	_, err := c.S3().CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(bucket),
+		CreateBucketConfiguration: &s3.CreateBucketConfiguration{
+			LocationConstraint: aws.String(c.Region),
+		},
+	})
+	if err != nil {
+		return errors.Wrap(err, "creating bucket", bucket)
+	}
+	return nil
+}
+
+// Checks bucket existence and accessibility with credentials
+func (c *Client) DoesBucketExist(bucket string) (bool, error) {
+	_, err := c.S3().HeadBucket(&s3.HeadBucketInput{
+		Bucket: aws.String(bucket),
```

Comment:
Did you confirm what will happen if the bucket exists but is in a different region (and you own it or you don't)?